internal/resolvers/default/load.go

```diff
@@ -72,7 +72,7 @@ func (dr *R) load(urlString string, r *http.Request) (interface{}, time.Duration
 		if err != nil {
 			return nil, cache.NoSpecialDur, err
 		}
-		if contentLengthBytes > resolver.MaxContentLength {
+		if contentLengthBytes > GetMaxContentLength() {
```

Comment:
You will need to multiply here by 1024 * 1024, so you compare bytes instead of mega bytes